### PR TITLE
[KED-2166] Add additional fields to meta data panel

### PR DESCRIPTION
**Pull request by bru5**
## Description

Adds additional fields to the metadata panel:

For parameter type nodes:
- file path
- parameters

For data type nodes:
- dataset type
- file path

For task type nodes:
- file path
- parameters (data appears when API finished)
- docstring

## Development notes

Parameters for function nodes will appear automatically once the backend API update is completed.

## QA notes

Check the above fields are correct for each type of node.

Test a real project through the proxy e.g. [spaceflights](https://github.com/quantumblacklabs/kedro-starter-spaceflights), `pip install` your local viz repo, then run `kedro viz --port 4142` alongside `npm run start` without any `api/main` file.

Any fields with missing or empty values will show as `-` (e.g. actually no value or not yet returned by API).

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: package.json
Comment (studioswong) [inline package.json:9]: I'm confused.... isn't this already merged in main? (https://github.com/quantumblacklabs/kedro-viz/pull/318)

**Comment by studioswong** [comment]
Thanks for the work @bru5 , good stuff. Please add some tests for the new node metadata fields that you have introduced for the `metadataRow` and `metadata` component to ensure they get displayed properly?

File: package.json
Comment (bru5) [reply package.json:9]: Yeah it is now, I just needed it here before it was merged a while back. Should just merge on top no problem I think.

**Comment by bru5** [comment]
For sure, I've expanded out the tests for each type now. The tests check the fields are shown, though some of them only return a blank value as the test data set needs updating. 

That part is not straightforward though because of the new additional API calls not yet existing in the test data approach, so I think we might need to check with @richardwestenra on how best to do that. 

Not ideal but don't think it needs to block this PR though, see what you think.

File: src/components/metadata/index.js
Comment (richardwestenra) [inline src/components/metadata/index.js:73]: Design question: Are we better off showing these fields regardless of whether they're populated, or only showing them if populated?
image

File: src/components/metadata/index.js
Comment (richardwestenra) [inline src/components/metadata/index.js:132]: image

Additional question: If we are going to always show these fields regardless of whether they're populated, as long as the node is the right corresponding type, then can we we please add dummy data to the demo dataset as well? We want the demo to always look as good as possible for obvious reasons.

Also, potentially the animals one as well? What do you think?

This can be part of a separate PR if you prefer.

**Review by richardwestenra — APPROVED** [review]

File: src/components/metadata/index.js
Comment (bru5) [reply src/components/metadata/index.js]: Yeah it's a good point and I did this intentionally, as I think it's important to show empty values in technical use cases like this. We can discuss and improve it later though if needed I think.

File: src/components/metadata/index.js
Comment (bru5) [reply src/components/metadata/index.js:133]: Yeah for sure the included data sets should be updated with the new fields. I was planning on getting your input on how to do that, as now we have two API calls, I wasn't sure how best to implement that. I think we should do it in a separate PR though from the functionality.

File: src/components/metadata/index.js
Comment (richardwestenra) [reply src/components/metadata/index.js]: Sweet as! cc @GabrielComymQB let's come back to this in 2021

File: src/components/metadata/index.js
Comment (richardwestenra) [reply src/components/metadata/index.js:133]: ok cool. yeah I think for the demo dataset we can definitely inline it. Once we add the code snippets it'll get trickier and we may need to revisit this whole data loading thing from the ground up

**Pull request merged**
